File: src/main/java/net/felsstudio/fels/parser/ast/Visitor.java
package main.java.net.felsstudio.fels.parser.ast;

/**
 *
 * @author felek
 */
public interface Visitor {
    
    void visit(ArrayExpression s);
    void visit(AssignmentExpression s);
    void visit(BinaryExpression s);
    void visit(BlockStatement s);
    void visit(BreakStatement s);
    void visit(ConditionalExpression s);
    void visit(ContainerAccessExpression s);
    void visit(ContinueStatement s);
    void visit(DoWhileStatement s);
    void visit(DestructuringAssignmentStatement s);
    void visit(ForStatement s);
    void visit(ForeachArrayStatement s);
    void visit(ForeachMapStatement s);
    void visit(FunctionDefineStatement s);
    void visit(FunctionReferenceExpression e);
    void visit(ExprStatement s);
    void visit(FunctionalExpression s);
    void visit(IfStatement s);
    void visit(ImportStatement s);
    void visit(MapExpression s);
    void visit(MatchExpression s);
    void visit(PrintStatement s);
    void visit(PrintlnStatement s);
    void visit(ReturnStatement s);
    void visit(TernaryExpression s);
    void visit(UnaryExpression s);
    void visit(ValueExpression s);
    void visit(VariableExpression s);
    void visit(WhileStatement st);
    void visit(UsingStatement st);
    void visit(PanicStatement st);
    void visit(ClassDeclarationStatement st);
    void visit(ObjectCreationExpression st);
}

File: src/main/java/net/felsstudio/fels/parser/ast/Visitor.kt
package main.java.net.felsstudio.fels.parser.ast

/**
 *
 * @author felek
 */
interface Visitor {
    fun visit(s: ArrayExpression?)
    fun visit(s: AssignmentExpression?)
    fun visit(s: BinaryExpression?)
    fun visit(s: BlockStatement?)
    fun visit(s: BreakStatement?)
    fun visit(s: ConditionalExpression?)
    fun visit(s: ContainerAccessExpression?)
    fun visit(s: ContinueStatement?)
    fun visit(s: DoWhileStatement?)
    fun visit(s: DestructuringAssignmentStatement?)
    fun visit(s: ForStatement?)
    fun visit(s: ForeachArrayStatement?)
    fun visit(s: ForeachMapStatement?)
    fun visit(s: FunctionDefineStatement?)
    fun visit(e: FunctionReferenceExpression?)
    fun visit(s: ExprStatement?)
    fun visit(s: FunctionalExpression?)
    fun visit(s: IfStatement?)
    fun visit(s: ImportStatement?)
    fun visit(s: MapExpression?)
    fun visit(s: MatchExpression?)
    fun visit(s: PrintStatement?)
    fun visit(s: PrintlnStatement?)
    fun visit(s: ReturnStatement?)
    fun visit(s: TernaryExpression?)
    fun visit(s: UnaryExpression?)
    fun visit(s: ValueExpression?)
    fun visit(s: VariableExpression?)
    fun visit(st: WhileStatement?)
    fun visit(st: UsingStatement?)
    fun visit(st: PanicStatement?)
    fun visit(st: ClassDeclarationStatement?)
    fun visit(st: ObjectCreationExpression?)
}
